### PR TITLE
Claudia 4724

### PR DESCRIPTION
**Pull request by chemaper**
#### Reviewers

@flopezag @jframos 
#### Description

Implementation of CLAUDIA 4724, about cmdline and unification of tools.

Most of the code was already integrated. The two operations added:
-make backups
-obtain the list of regions

The corresponding scripts in examples has been removed.

The other changes in the cmdline:
-Add an option to override any option in the configuration file. This is more powerful and clean than adding specific options in the cmdline for some operations
-Support "target:" to expand to the list of regions in that target.

The documentation is now more completed, with information about these changes and also about the mock used for testing.
#### Testing

Locally

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -638,6 +719,36 @@ change testingFacadeReal to True. It needs the usual OpenStack environment
...
+* new_scenario: there are files for each region with the backup of the metadata
+                BEFORE invoking the synchronisation. These files can be generated
+                with *sync.py --make-backup*
+* new_scenario.result: there are files for each region with teh backup of the
+                       metadata AFTER invoking the synchronisation
+* new_scenario_pre: there are files with the status of each region BEFORE invoking
+                    the synchronisation. These files can be generated with the
+                    output of *sync.py --show-status*
+* new_scenario_post: there are files with the status of each region AFTER invoking
+                     the synchronisation. These files can be generated with the
```
Comment (jframos) [inline README.rst:737]: with what?

Comment (chemaper) [inline README.rst:737]: You are right. Added missing line and also a paragraph about the optional config file.

**Comment by jframos** [comment]
It would be very interesting installing GlanceSync as a 'system' command, when executing `pip install GlanceSync` (like the PolicyManager components).

Perhaps we could upload the component to pypi repository!

File: sync.py [hunk 2]
```diff
@@ -159,6 +175,43 @@ def _wait_child(self, children):
...
+    def show_regions(self):
+        """print a full list of the regions available (excluding the
+        master region) in all the targets defined in the configuration file"""
+        regions = self.glancesync.get_regions()
+        for target in self.glancesync.targets.keys():
+            if target == 'facade' or target == 'master':
+                continue
+            regions.extend(self.glancesync.get_regions(target=target))
+
+        print ' '.join(regions)
```
Comment (flopezag) [inline sync.py]: try to change it to:

print(' '.join(regions))

**Comment by flopezag** [comment]
@jframos: It would be very interesting installing GlanceSync as a 'system' command, when executing pip install GlanceSync (like the PolicyManager components). Perhaps we could upload the component to pypi repository!

This is clearly other user story for the next sprint.

**Comment by jframos** [comment]
:+1: 
LGTM

**Comment by flopezag** [comment]
LGTM

**Pull request merged**